src/neo/Network/P2P/LocalNode.cs

```diff
@@ -129,6 +131,47 @@ internal static IPEndPoint GetIpEndPoint(string hostAndPort)
             return null;
         }
 
+        /// <summary>
+        /// Check duplicated duplicated Nonce. Usually it occurs when a new remote connection is established, which checks its counterpart's Nonce value. <br/>
+        /// If Nonce is the same we check if the Remote can be added to the known LocalAddresses.<br/>
+        /// If it is equal to the Nonce of other RemoteNode, we just return true, else we'll return false and update the Listener address of the connected remote node.
+        /// </summary>
+        /// <param name="remoteActor">Remote node actor</param>
+        /// <param name="remoteNode">Remote node</param>
+        public bool CheckDuplicateNonce(IActorRef remoteActor, RemoteNode remoteNode)
+        {
+            var version = remoteNode.Version;
+            var remote = remoteNode.Remote;
+
+            if (remote is null)
+            {
+                return false;
+            }
+            if (version.Nonce == Nonce)
+            {
+                if (LocalAddresses.Count < MaxConnections)
+                {
+                    LocalAddresses.Add(remote.Address);
```

Comment:
The nonce could be the same even if they are not the same node. So you can't add the address to the `LocalAddresses` list.

Reply:
It happens very rarely, and acceptable.

Reply:
This line should be removed.

Reply:
Done